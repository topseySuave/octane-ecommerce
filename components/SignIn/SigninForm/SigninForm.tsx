import { useState } from "react";
import Router from "next/router";
import { Input, Icon, Button, Typography } from "antd";
import WithSocialMedia from "components/WithSocialMedia";
import { Link } from "lib/routes";

const SigninForm = () => {
  const [loading, setLoading] = useState(false);

  const login = () => {
    setLoading(true);
    setTimeout(() => {
      setLoading(false);
      Router.push("/");
    }, 5000);
  };

  return (
    <>
      <form>
        <Input
          placeholder="Enter your Email"
          type="email"
          prefix={<Icon type="user" style={{ color: "rgba(0,0,0,.25)" }} />}
        />
        <Input.Password
          prefix={<Icon type="lock" style={{ color: "rgba(0,0,0,.25)" }} />}
          placeholder="input password"
        />
        <Button type="primary" loading={loading} onClick={login} block>
          Sign In
        </Button>
      </form>
      <Typography.Text style={{ textAlign: "center" }}>
        Don't have an account?{" "}
        <Link prefetch href="/signup">
          <a>sign up</a>
        </Link>
      </Typography.Text>
      <WithSocialMedia />
    </>
  );
};

export default SigninForm;
