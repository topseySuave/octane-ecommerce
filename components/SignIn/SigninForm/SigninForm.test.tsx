import { shallow } from 'enzyme';
import React from 'react';

import SigninForm from './SigninForm';

describe('SigninForm', () => {
  it('Renders with sign in form', () => {
    const tree = shallow(<SigninForm />);
    expect(tree).toMatchSnapshot();
    // const mounted = shallow(<Profile />);
    // const profile = mounted.find('content');
    // console.log(profile);
  });
});
