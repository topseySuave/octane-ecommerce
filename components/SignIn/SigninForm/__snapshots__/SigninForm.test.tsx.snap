// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SigninForm Renders with sign in form 1`] = `ShallowWrapper {}`;
